pyccel/parser/base.py

```diff
@@ -231,6 +232,7 @@ def __init__(self,
         self._metavars  = OrderedDict()
         self._namespace = Scope()
 
+        self._used_names = None
```

Comment:
Given that you changed this class, I think this is a good opportunity to update its docstring, which has not been updated in a while!

Reply:
I am not sure exactly what to add to this docstring. I have added a short description

Reply:
The docstring should be removed from the `__init__` and added to the class docstring, and we should add a description for `output_folder`...

Reply:
The `output_folder` property does not seem to be used. I will remove it

Reply:
Great, thanks!